module/nvpair/nvpair.c

```diff
@@ -3251,8 +3251,12 @@ nvs_xdr_nvpair(nvstream_t *nvs, nvpair_t *nvp, size_t *size)
 			return (0);
 
 		/* sanity check the size parameter */
+#ifndef WITH_TIRPC
 		if (!xdr_control(xdr, XDR_GET_BYTES_AVAIL, &bytesrec))
 			return (EFAULT);
+#else
+		xdr_control(xdr, XDR_GET_BYTES_AVAIL, &bytesrec);
```

Comment:
libtirpc's implementation of xdr_control appears to be a no-op because it checks for a function pointer that is always NULL. That means that `bytesrec` is uninitialized before it is accessed by `NVS_XDR_MAX_LEN`.

It looks like our `xdr_control` would work here, so we probably should undefine the libtirpc xdr_control in favor of that. We could do that by doing something like this in our rpc/xdr.h:

```
#include_next <rpc/xdr.h>

/* libtirpc xdr_control is unimplemented as of 0.2.5. */
#ifdef WITH_TIRPC
#ifdef xdr_control
#undef xdr_control
#endif
#endif
```